### PR TITLE
Allow users to specify annotation in pod if only local node should be…

### PR DESCRIPTION
**Pull request by disrani-px**
… preferred

**What type of PR is this?**
Improvement

**What this PR does / why we need it**:

**Does this PR change a user-facing CRD or CLI?**:
No

**Is a release note needed?**:
```release-note
Users can now set the annotation `stork.libopenstorage.org/preferLocalNodeOnly` to `true` in their pods if they want pods to be scheduled only on nodes where data is located.
```

**Does this change need to be cherry-picked to a release branch?**:
2.2

### CODE REVIEW

File: pkg/extender/extender.go [hunk 1]
```diff
@@ -176,30 +178,58 @@ func (e *Extender) processFilterRequest(w http.ResponseWriter, req *http.Request
...
+					storklog.PodLog(pod).Errorf("No online storage nodes have replica for volume, returning error")
 					msg := "No online node found with volume replica"
 					e.Recorder.Event(pod, v1.EventTypeWarning, schedulingFailureEventReason, msg)
 					http.Error(w, msg, http.StatusBadRequest)
 					return
 				}
 			}
 
+			preferLocalOnly := false
+			if value, ok := pod.Annotations[preferLocalNodeOnlyAnnotaion]; ok {
```
Comment (harsh-px) [inline pkg/extender/extender.go]: safe to nil check `pod.Annotations`

File: pkg/extender/extender.go [hunk 2]
```diff
@@ -34,9 +35,10 @@ const (
 	// same region as a node which has data for the volume
 	regionPriorityScore = 10
 	// defaultScore Score assigned to a node which doesn't have data for any volume
-	defaultScore = 5
-
+	defaultScore                 = 5
 	schedulingFailureEventReason = "FailedScheduling"
+	// annotation to check if only local nodes should be used to schedule a pod
+	preferLocalNodeOnlyAnnotaion = "stork.libopenstorage.org/preferLocalNodeOnly"
```
Comment (harsh-px) [inline pkg/extender/extender.go]: type in `preferLocalNodeOnlyAnnotaion`

File: pkg/extender/extender.go (hunk 1 above)
Comment (disrani-px) [reply pkg/extender/extender.go]: Done

File: pkg/extender/extender.go (hunk 2 above)
Comment (disrani-px) [reply pkg/extender/extender.go]: Done

**Pull request merged**